core/src/replay_stage.rs

```diff
@@ -3639,15 +3640,27 @@ impl ReplayStage {
                     }
                 }
 
+                let is_leader_block = bank.collector_id() == my_pubkey;
```

Comment:
seems like this whole thing (3643 - 3691) could be hidden behind some `determine_and_set_block_id` function

Reply:
yeah fair, factored it out here https://github.com/anza-xyz/alpenglow/pull/621/commits/dc7153e8166a711729dad845a45accc06d7fe49f